File: open_root_files.py
from time import time

import ROOT

def open_files(filenames):
    return [ROOT.TFile.Open(filename) for filename in filenames]

filenames = [("root://eospublic.cern.ch/"
              "/eos/root-eos/benchmark/CMSOpenDataDimuon/"
              "Run2012BC_DoubleMuParked_Muons_{}.root").format(i)
             for i in range(1, 101)
             ]

if __name__ == "__main__":

    with open("benchmarks/open_root_files.csv", "a+") as f:
        for _ in range(5):
            print("Opening remote files...")
            start = time()
            files = open_files(filenames)
            end = time()

            f.write("{}\n".format(round(end - start, 2)))

File: open_root_files.cpp
#include <fstream>
#include <iomanip> // std::setprecision
#include <iostream>
#include <string>

#include <TFile.h>
#include <TStopwatch.h>

int main()
{
    static std::string filenametemplate{"root://eospublic.cern.ch//eos/root-eos/benchmark/CMSOpenDataDimuon/Run2012BC_DoubleMuParked_Muons_"};
    static std::string outcsv{"benchmarks/open_root_files_cpp.csv"};
    std::ofstream timecsv{outcsv.c_str(), std::ofstream::out | std::ofstream::app};

    for(int j = 0; j < 10; j++){
        TStopwatch watch;
        for (int i = 1; i < 101; i++)
        {
            std::string filename{filenametemplate + std::to_string(i) + ".root"};
            TFile::Open(filename.c_str());
        }
        double elapsed{watch.RealTime()};

        // Store elapsed time in a csv file
        std::cout << "Files opened in: " << std::fixed << std::setprecision(2) << elapsed << " s\n";

        timecsv << std::fixed << std::setprecision(2) << elapsed << "\n";
    }
    timecsv.close();
}
